R/mainTests.r

```diff
@@ -1153,6 +1160,454 @@ country_code <- function(data) {
+                             <br> - Each geography should have only one code."),
+            "result" = "FAIL"
+          )
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+# other_geography_code_duplicates  ----------------------------------------
+# check that there is a 1:1 relationship between geography names and codes
+
+lower_level_geog_names <- geography_matrix[c(7:12, 14:16), 2:3] %>% as.character()
+
+other_geography_code_duplicates <- function(data) {
+  if (!any(lower_level_geog_names %in% names(data))) {
+    output <- list(
+      "message" = "Lower-level geography data is not present in this data file.",
+      "result" = "IGNORE"
+    )
+  } else {
+    geog_data <- data %>%
+      select(any_of(c(
+        "geographic_level", lower_level_geog_names
+      ))) %>%
+      distinct() %>%
+      mutate(ID = 1:n())
+
+    names <- geog_data %>%
+      select(ID, geographic_level, contains("name")) %>%
+      mutate_if(is.numeric, as.character) %>%
+      melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+      select(ID, geographic_level, name = value)
+
+    codes <- geog_data %>%
+      select(ID, geographic_level, !contains("name")) %>%
+      mutate_if(is.numeric, as.character) %>%
+      melt(id.vars = c("ID", "geographic_level"), na.rm = TRUE) %>%
+      select(ID, geographic_level, code = value)
+
+    lookup_creator <- names %>%
+      full_join(codes, by = c("ID", "geographic_level")) %>%
+      select(-ID, -geographic_level) %>%
+      distinct() %>%
+      add_count(name, name = "name_n") %>%
+      add_count(code, name = "code_n")
+
+    multi_count_code <- lookup_creator %>%
+      filter(code_n > 1) %>%
+      mutate(combo = paste0(code, " - ", code_n, " different names")) %>%
+      select(combo) %>%
+      distinct() %>%
+      pull()
+
+    if (length(multi_count_code) == 0) {
+      output <- list(
+        "message" = "every geography code has one assigned geography.",
+        "result" = "PASS"
+      )
+    } else {
+      if (length(multi_count_code) == 1) {
+        output <- list(
+          "message" = paste0("The following geography code has multiple assigned geographies: '", paste0(multi_count_code), "'. 
+                             <br> - Each geography code should have only one assigned geography."),
+          "result" = "FAIL"
+        )
+      } else {
+        if (length(multi_count_code) > 1) {
+          output <- list(
+            "message" = paste0("The following geography codes have multiple assigned geographies: '", paste0(multi_count_code, collapse = "', '"), "'.
+                             <br> - Each geography code should have only one assigned geography."),
+            "result" = "FAIL"
+          )
+        }
+      }
+    }
+  }
+  return(output)
+}
+
+# na_geography -------------------------------------
+# checking if location has code of ":", then name is "not available"
+
+geography_name_codes <- geography_matrix[, 2:3] %>% as.character()
```

Comment:
Think you can add `%>% .[!is.na(.)]` to line 1494 and remove the need for line 1496?

Reply:
so much neater thanks!